src/System.Private.CoreLib/shared/System/Convert.cs

```diff
@@ -2518,21 +2520,131 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
                 charsWritten = ConvertToBase64Array(outChars, inData, 0, bytes.Length, insertLineBreaks);
                 return true;
             }
+        }      
+
+        internal static readonly Vector128<byte> s_base64ShuffleMask = Vector128.Create((byte)
+            1, 0, 2, 1, 4, 3, 5, 4, 7, 6, 8, 7, 10, 9, 11, 10);
+
+        internal static readonly Vector128<byte> s_base64ShiftLut = Vector128.Create(
+            (sbyte)'a' - 26, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'0' - 52,
+            (sbyte)'0' - 52, (sbyte)'+' - 62,
+            (sbyte)'/' - 63, (sbyte)'A', 0, 0).AsByte();
+
+        internal static readonly Vector128<byte> s_base64TwoBytesStringMaskLo = Vector128.Create(
+                0, 0x80, 1, 0x80,
+                2, 0x80, 3, 0x80,
+                4, 0x80, 5, 0x80,
+                6, 0x80, 7, 0x80);
+
+        private static unsafe (int i, int j, int charcount) ConvertToBase64ArraySsse3(char* outChars, byte* inData, int length, int offset, bool insertLineBreaks)
+        {
+            int i = offset, j = 0, charcount = 0;
+            const int stride = 4 * 3;
+
+            // Based on "Base64 encoding with SIMD instructions" article by Wojciech Muła	
+            // http://0x80.pl/notesen/2016-01-12-sse-base64-encoding.html
+            byte* outputBytes = (byte*)outChars;
+
+            Vector128<byte>   tt0 = Vector128.Create(0x0fc0fc00).AsByte();
+            Vector128<ushort> tt1 = Vector128.Create(0x04000040).AsUInt16();
+            Vector128<byte>   tt2 = Vector128.Create(0x003f03f0).AsByte();
+            Vector128<ushort> tt3 = Vector128.Create(0x01000010).AsUInt16();
+            Vector128<byte>   tt5 = Vector128.Create((byte)51);
+            Vector128<sbyte>  tt7 = Vector128.Create((sbyte)26);
+            Vector128<byte>   tt8 = Vector128.Create((byte)13);
+
+            Vector128<byte> localShiftLut = s_base64ShiftLut;
+            Vector128<byte> localShuffleMask = s_base64ShuffleMask;
+            Vector128<byte> localTwoBytesStringMaskLo = s_base64TwoBytesStringMaskLo;
+
+            for (; i <= length - stride; i += stride)
+            {
+                Vector128<byte> inputVector = Sse2.LoadVector128(inData + i);
```

Comment:
nit: Would be nice to have the comment that you load: `[xxxx|WWWZ|ZZYY|YXXX]` and then shuffle it to `[WWWW|ZZZZ|YYYY|XXXX]`

Reply:
and that the overall layout is: `[bbbbcccc|CCdddddd|aaaaaaBB|bbbbcccc]`

Reply:
I copied the comments from the C implementation: https://github.com/WojciechMula/base64simd/blob/master/encode/encode.sse.cpp#L20-L59
(I've just added the comments for Load + Shuffle)